simulation.py

```diff
@@ -1,293 +1,524 @@
+
+
+        #Now that the SimulationThread has been created (but not yet running), connect the
+        # Button clicked in MainWindow - call a SimulationThread method to do something
+        self.buttonRun.clicked.connect(self.simulationThread.thread_start_calculating)
+        self.buttonStop.clicked.connect(self.simulationThread.thread_stop_calculating)
+        
+        self.simulationThread.start()
+        self.plotRefreshThread.start()
+        
+    def createControls(self):
+        self.labelStatus = QLabel("Status")
+        self.textboxStatus = QLineEdit("Initialized",self)
+        self.textboxStatus.setReadOnly(True)
+        self.buttonRun = QPushButton('Run/Continue', self)
+        self.buttonRun.setEnabled(True) 
+        self.buttonStop = QPushButton('Pause', self)
+        self.buttonStop.setEnabled(True) 
+        
+        #a spacer to add visual space at the bottom of the gui (not used yet)
+        #controlVerticalSpacer = QtGui.QSpacerItem(40,20, QSizePolicy.Minimum, QSizePolicy.Expanding)
+        self.controlsGroup = QtGui.QGroupBox('Controls')
+        self.controlsLayout= QtGui.QGridLayout()
+        self.controlsLayout.addWidget(self.labelStatus,     0, 1)
+        self.controlsLayout.addWidget(self.textboxStatus,   0, 2)
+        self.controlsLayout.addWidget(self.buttonRun,       0, 3)
+        self.controlsLayout.addWidget(self.buttonStop,      0, 4)
+        #self.controlsLayout.addItem(controlVerticalSpacer, 0, 3,QtCore.Qt.AlignTop)
+        self.controlsGroup.setLayout(self.controlsLayout)
+        
+
+    def simulationThreadResultsDataDisplay(self):
+        # TBD placeholder for real work to be done when the SimulationThread (a simulationThread thread)
+        # SIGNALs MainWindow new data is available in shared memory
+        print('Window SIGNAL from SimulationThread: Results_data_ready') 
+
+    
+    def simulationThreadFinished(self):
+        # TBD placeholder for SimulationThread SIGNALs ??exiting
+        # data is available in shared memory
+        print('Window: SIGNAL From SimulationThread: Finished') 
+
+    def simulationThreadTerminated(self):
+        # TBD placeholder for SimulationThread SIGNALs terminated
+        print('Window: SIGNAL From SimulationThread: Terminated') 
+
+    ###################################
+    #TBD REMOVE/REPLACE FOR THE SIM APP
+    @pyqtSlot(str)
+    def signalRcvFromSimulationThread(self, text ):
+        #self.buttonRun.setText(text)
+        self.textboxStatus.setText(text)
+
+        
+    @pyqtSlot()
+    def signalPlotRefresh(self):
+        index = self.data_store.get_simulation_index()
+        print("MainWindow:signalPlotRefresh Simulation Index = {}".format(index))
+        # create a new plot for every point simulated so far
+        x = [z+1 for z in range(self.data_store.get_simulation_index())]
+        _velocity = []
+        _distance = []
+        _battery_power = []
+        for z in x:
```

Comment:
Add todo to move getting the velocity results from the beginning to the current index into the datastore, I think building the lists in the datastore will be faster (less locking/unlocking of thread lock)